packages/generator/tests/acceptance/test_pin_aware_measurement_client.py

```diff
@@ -0,0 +1,197 @@
+
+
+def test___measurement_plugin_client___measure_with_default_site_selection___returns_selected_site(
+    measurement_plugin_client_module: ModuleType,
+    pin_map_directory: pathlib.Path,
+) -> None:
+    pin_map_path = pin_map_directory / "1Smu1ChannelGroup1Pin1Site.pinmap"
+    test_measurement_client_type = getattr(measurement_plugin_client_module, "TestMeasurement")
+    measurement_plugin_client = test_measurement_client_type()
+    measurement_plugin_client.register_pin_map(pin_map_path)
+
+    output = measurement_plugin_client.measure()
+
+    assert output.sites == [0]
+
+
+def test___measurement_plugin_client___measure_with_multiple_sites_selection___returns_selected_sites(
+    measurement_plugin_client_module: ModuleType,
+    pin_map_directory: pathlib.Path,
+) -> None:
+    pin_map_path = pin_map_directory / "1Smu1ChannelGroup2Pin2Site.pinmap"
+    test_measurement_client_type = getattr(measurement_plugin_client_module, "TestMeasurement")
+    measurement_plugin_client = test_measurement_client_type()
+    measurement_plugin_client.register_pin_map(pin_map_path)
+
+    measurement_plugin_client.sites = [0, 1]
+    output = measurement_plugin_client.measure()
+
+    assert output.sites == [0, 1]
+
+
+def test___measurement_plugin_client___measure_with_invalid_sites_selection___raises_invalid_sites_error(
+    measurement_plugin_client_module: ModuleType,
+    pin_map_directory: pathlib.Path,
+) -> None:
+    pin_map_path = pin_map_directory / "1Smu1ChannelGroup1Pin1Site.pinmap"
+    test_measurement_client_type = getattr(measurement_plugin_client_module, "TestMeasurement")
+    measurement_plugin_client = test_measurement_client_type()
+    measurement_plugin_client.register_pin_map(pin_map_path)
+
+    with pytest.raises(grpc.RpcError) as exc_info:
+        measurement_plugin_client.sites = [0, 1]
+        _ = measurement_plugin_client.measure()
+
+    assert exc_info.value.code() == grpc.StatusCode.UNKNOWN
+    assert 'Pin map does not contain site numbers: "1"' in (exc_info.value.details() or "")
+
+
+def test___measurement_plugin_client___measure_with_pin_map_context___returns_output(
+    measurement_plugin_client_module: ModuleType,
+    pin_map_directory: pathlib.Path,
+) -> None:
+    pin_map_path = pin_map_directory / "1Smu1ChannelGroup2Pin2Site.pinmap"
+    output_type = getattr(measurement_plugin_client_module, "Output")
+    expected_output = output_type(
+        pin_map_id=str(pin_map_path),
+        sites=[0, 1],
+        session_names=["DCPower1/0, DCPower1/1, DCPower1/2, DCPower1/3"],
+        resource_names=["DCPower1/0, DCPower1/1, DCPower1/2, DCPower1/3"],
+        channel_lists=["DCPower1/0, DCPower1/2"],
+    )
+    test_measurement_client_type = getattr(measurement_plugin_client_module, "TestMeasurement")
+    measurement_plugin_client = test_measurement_client_type()
+    measurement_plugin_client.register_pin_map(pin_map_path)
```

Comment:
Is this register_pin_map call required?

Reply:
Thanks for pointing this out. register pin map call is not needed here, it negates the point of this test.